app/support/DbAdapter/users.js

```diff
@@ -712,6 +713,21 @@ const usersTrait = (superClass) =>
         { sparseQuery: `%${query.split('').join('%')}%`, query },
       );
     }
+
+    /**
+     * @param {UUID} userId
+     * @returns {Promise<Date|null>}
+     * @this {DbAdapter}
+     */
+    async setFirstUserInteraction(userId) {
+      const pool = await this.getSlonik();
+      const tSql = sql.type(z.object({ first_interaction_at: z.date() }));
+      return await pool.maybeOneFirst(
+        tSql`update users set first_interaction_at = now() 
+          where first_interaction_at is null and type = ${'user'} and uid = ${userId}
```

Comment:
I don't think you need to use prepared statement for string literal. Just pass it directly

```suggestion
          where first_interaction_at is null and type = 'user' and uid = ${userId}
```